### PR TITLE
BAPI - mapmanip submap area/turf handling and submap repeating

### PR DESCRIPTION
**Pull request by DreamySkrell**
changes:
  - rscadd: "BAPI - mapmanip submap noop handling."
  - rscadd: "BAPI - mapmanip submap repeat setting."

both features are explained in forum thread <https://forums.aurorastation.org/topic/20511-mapmanip-guide-submaps/>

but basically:
- "noop" areas and turfs has special meaning and determine whether the submap manipulation is more like "replacing" or "appending"
- repeat setting controls whether extracted submaps should repeat on insertion

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!build_bapi

**Comment by DreamySkrell** [comment]
!build_bapi

**Comment by DreamySkrell** [comment]
!build_bapi

**Comment by DreamySkrell** [comment]
!review

**Review by Arrow768 — APPROVED** [review]

File: rust/bapi/src/mapmanip/tools/insert_submap.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    // src tile has neither noop area nor turf
+                    // the simplest case, just replace the whole dst tile with the src tile
+                    // use src turf, and src area
+                    tile_dst.prefabs = tile_src.prefabs;
+                    tile_dst.prefabs.push(tile_src_turf);
+                    tile_dst.prefabs.push(tile_src_area);
+                    // do note that in dmm file format
+                    // turf and area have to be the two last elements in a prefab
+                }
+            }
```
Comment (Llywelwyn) [inline rust/bapi/src/mapmanip/tools/insert_submap.rs:84]:
```rust
let new_area = if tile_src_area_is_noop { tile_dst_area } else { tile_src_area };
let new_turf;
if tile_src_turf_is_noop {
	new_turf = tile_dst_turf
	tile_dst.prefabs.append(&mut tile_src.prefabs)
} else {
	new_turf = tile_src_turf
	tile_dst.prefabs = tile_src.prefabs
}
tile_dst.prefabs.push(new_turf);
tile_dst.prefabs.push(new_area);
```
something like this is functionally the same but a little cleaner imo

Comment (DreamySkrell) [reply rust/bapi/src/mapmanip/tools/insert_submap.rs]: done, thanks

**Comment by DreamySkrell** [comment]
!build_bapi

**Comment by DreamySkrell** [comment]
!review

**Pull request merged**